### DISCUSSION TITLE
why does my @click not work (total noob q - sorry)

### DISCUSSION THREAD

**Discussion by wdiechmann** [alpinejs/alpine#614] [4. General]
please find my demo below.

I'd expect to see an alert with 'hi1' first, then a second with 'hi2', and finally a second 'hi2' once I tap the H1'ed Dashboard text - but all I get is the initial (not alpine) 'Hi1'

I'm totally new to AlpineJS - what am I doing wrong?

Can someone please tell me that? 

```
<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8"/>
    <meta http-equiv="X-UA-Compatible" content="IE=edge"/>
    <meta name="viewport" content="width=device-width, initial-scale=1.0"/>
    <title>AlpineJS demo</title>
    <link rel="stylesheet" href="/css/app.css" />
    <script src="https://cdn.jsdelivr.net/gh/alpinejs/alpine@v2.x.x/dist/alpine.min.js" defer></script>
    <script type="text/javascript" src="/js/app.js"></script>
  </head>
  <body>
    <div class="py-10">
      <header>
        <div class="max-w-7xl mx-auto px-4 sm:px-6 lg:px-8">
          <h1 @click="alertJS" class="text-3xl font-bold leading-tight text-gray-900">
            Dashboard
          </h1>
        </div>
      </header>
      <main x-init="() => { alertJS() }">
        <div class="max-w-7xl mx-auto sm:px-6 lg:px-8">
          <div class="px-4 py-8 sm:px-0">
            <div class="border-4 border-dashed border-gray-200 rounded-lg h-96"></div>
          </div>
        </div>
      </main>
    </div>
  </div>
  <script>
    window.alertJS = function (){
      alert('hi2')
    }

    function r(f){/in/.test(document.readyState)?setTimeout('r('+f+')',9):f()}

    r(function(){
      alert('hi1')
    });

  </script>
  </body>
</html>
```

**Comment by ryangjchandler** [comment]
@wdiechmann You've not got an `x-data` attribute anywhere.

**Reply by wdiechmann** [reply]
ohh - so an x-data is required to get anything going? got it! The docs just does not stress that as a requirement - as far as I can see 😦

**Comment by wdiechmann** [comment]
hi @ryangjchandler - great of you to point me in the right direction!

Now my x-init works - but your  'trick' didn't help me on the `h1` I'm afraid 😞 
Added an x-data to my `<h1>` tag but I cannot click the headline - is there something else I need to add?

```
  <header>
        <div class="max-w-7xl mx-auto px-4 sm:px-6 lg:px-8">
          <h1 x-data="{ alert2: false }" @click="alertJS" class="text-3xl font-bold leading-tight text-gray-900">
            Dashboard
          </h1>
        </div>
      </header>
      <main x-data="{ alert1: false }" x-init="() => { alertJS() }">
        <div class="max-w-7xl mx-auto sm:px-6 lg:px-8">
          <div class="px-4 py-8 sm:px-0">
            <div class="border-4 border-dashed border-gray-200 rounded-lg h-96"></div>
          </div>
        </div>
      </main>
 
```

**Reply by SimoTod** [reply]
`@click="alertJS()"`
See https://codepen.io/SimoTod/pen/BajwWOX
Note, you can use empty x-data directoves if you don't have any parameters for your component.

**Reply by wdiechmann** [reply]
super - thx! Should have thought about that - but was mislead by the examples on x-on which do not stress the importance of the ()'s

thx again for your clear-headed overview, time and attention!

**Comment by SimoTod** [comment]
No worries. Parentheses are optional only for functions defined in x-data.
Despite Alpine being able to use functions defined on your global scope,
the idea is that you try to encapsulate most of not all your logic in a
component (x-data).

As a separate note, you can find more resources / examples here ->
https://github.com/alpinejs/awesome-alpine

They should help you understand how Alpine works.
